data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/AddEntryProcessorConfig.java

```diff
@@ -46,21 +50,36 @@ public String getFormat() {
             return format;
         }
 
+        public String getValueExpression() {
+            return valueExpression;
+        }
+
         public boolean getOverwriteIfKeyExists() {
             return overwriteIfKeyExists;
         }
 
         public String getAddWhen() { return addWhen; }
 
-        @AssertTrue(message = "Either value or format must be specified")
-        public boolean hasValueOrFormat() {
-            return Objects.nonNull(value) || Objects.nonNull(format);
+        @AssertTrue(message = "Either value or format or expression must be specified, and only one of them can be specified")
+        public boolean hasValueOrFormatOrExpression() {
+            int numValuesConfigured = 0;
+            if (Objects.nonNull(value)) {
+                numValuesConfigured++;
+            }
+            if (Objects.nonNull(format)) {
+                numValuesConfigured++;
+            }
+            if (Objects.nonNull(valueExpression)) {
+                numValuesConfigured++;
+            }
+            return numValuesConfigured == 1;
```

Comment:
You can simplify this to the following example using java's XOR operator. Then there is no need to create an int, mutate and compare at the end.: 

```
return Objects.nonNull(value) ^ Objects.nonNull(format) ^ Objects.nonNull(valueExpression);
```

Reply:
Oh wait, disregard that. If all values are nonNull this would pass as well.